etc/vcl_snippets/recv.vcl

```diff
@@ -152,10 +152,12 @@
     unset req.http.graphql;
     # GraphQL special headers handling because this area doesn't rely on X-Magento-Vary cookie
     if (req.request == "GET" && req.url.path ~ "/graphql" && req.url.qs ~ "query=") {
-        if ( req.http.Authorization ~ "^Bearer" ) {
+        set req.http.graphql = "1";
+        if ( req.http.X-Magento-Cache-Id ) {
+            unset req.http.X-Magento-Vary;
+        } else if ( req.http.Authorization ~ "^Bearer" ) {
             set req.http.x-pass = "1";
```

Comment:
Does this mean that we are still bypassing the cache for queries with authorization token?

Reply:
If X-Magento-Cache-Id is not sent, yes.

Reply:
I take it the app is adding X-Magento-Cache-Id request header ?

Reply:
Yes.  For GraphQL requests, we want to replace the X-Magento-Vary cookie with an X-Magento-Cache-Id header which will be generated by the backend and echoed back by the frontend on subsequent requests.